File: dog.rb
class Dog

	def initialize(args)
		@name = args[:name]
		@breed = args[:breed]
		@age = args[:age]
	end

  def bark
    puts "Woof woof"
  end
  
end

dog = Dog.new({name: "Chancy", breed: "Dalmation", age: 11})
dog.bark

File: dog.js
function Dog(args){
	this.name = args["name"] || "shoes";
	this.breed = args.breed;
	this.age = args.age;

	this.bark = function(){
		return this.name;
	}

}


// Dog.prototype.bark = function(){
// 	return "Ruff ruff";
// }

var dog = new Dog({breed:"Red Panda", age:87});

console.log(dog.bark());
